lib/utils/wechat-mp.js

```diff
@@ -0,0 +1,164 @@
+    }
+    // fix section
+    $('section').each((_, section) => {
+        section = $(section);
+        const p_count = section.find('p').length;
+        const div_count = section.find('div').length;
+        const section_count = section.find('section').length;
+        if (p_count + div_count + section_count === 0) {
+            // make it a p
+            replaceTag($, section, 'p');
+        } else {
+            // make it a div
+            replaceTag($, section, 'div');
+        }
+    });
+    // clean scripts
+    $('script').remove();
+    return $.html();
+};
+
+/**
+ * Fetch article and its metadata from WeChat MP (mp.weixin.qq.com).
+ *
+ * If you use this function, no need to call `fixArticleContent`
+ * @param {object} ctx - The context object.
+ * @param {string} url - The url of the article.
+ * @param {boolean} bypassHostCheck - Whether to bypass host check.
+ * @return {Promise<object>} - An object containing the article and its metadata.
+ */
+const fetchArticle = async (ctx, url, bypassHostCheck = false) => {
+    const oriUrl = url;
+    url = url.replace(/^http:\/\//, 'https://').replace(/#\w*$/, ''); // normalize url
+    if (!bypassHostCheck && !url.startsWith('https://mp.weixin.qq.com/')) {
+        throw new Error('wechat-mp: URL must start with https://mp.weixin.qq.com/ or http://mp.weixin.qq.com/, but got ' + oriUrl);
+    }
+    return await ctx.cache.tryGet(url, async () => {
+        const response = await got(url);
+        const $ = cheerio.load(response.data);
+
+        const title = $('meta[property="og:title"]').attr('content');
+        const author = $('meta[name=author]').attr('content');
+        let summary = $('meta[name=description]').attr('content');
+        summary = summary !== title ? summary : '';
+        const description = fixArticleContent($('div#js_content.rich_media_content'));
+
+        let pubDate;
+        const publish_time_script = $('script[nonce][type="text/javascript"]:contains("var ct")').first().html();
+        const publish_time_match = publish_time_script && publish_time_script.match(/var ct *= *"?(\d{10})"?/);
+        const publish_timestamp = publish_time_match && publish_time_match[1];
+        if (publish_timestamp) {
+            pubDate = parseDate(publish_timestamp * 1000);
+        }
+
+        let mpName = $('.profile_nickname').first().text();
+        mpName = mpName && mpName.trim();
+
+        return { title, author, description, summary, pubDate, mpName, link: url };
+    });
+};
+
+/**
+ * Fetch article and its metadata from WeChat MP (mp.weixin.qq.com), then fill the `item` object with the result.
+ *
+ * If you use this function, no need to call `fetchArticle` or `fixArticleContent`
+ *
+ * A new route SHOULD use this function instead of manually calling the above functions
+ *
+ * An existing route adopting this function SHOULD either:
+ * - set `skipLink` to true (not recommended)
+ * - set `item.guid` to `item.link` BEFORE calling this function
```

Comment:
Could you explain why it's necessary to set `item.guid` to `item.link`? Middleware will handle it if no guid is set and guid isn't used in this utility.

https://github.com/DIYgod/RSSHub/blob/e0b7ca676d8b5817b474861ad674205444f3a15e/lib/views/rss.art#L35
https://github.com/DIYgod/RSSHub/blob/e0b7ca676d8b5817b474861ad674205444f3a15e/lib/views/atom.art#L37

Reply:
Only existing routes need to do that in order not to break existing subscriptions.

Different guid means different posts. RSS readers use guid to uniquely distinguish posts. A new guid will make the RSS reader consider the post as a never-seen post. Thus, it is necessary to make sure not to change the guid of the post of existing routes.

Reply:
This utility will normalize the URL. If not manually setting guid to the original URL, the guid will be changed and make RSS readers consider an ever-seen post as never-seen.

Reply:
Understood.